libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestClient.java

```diff
@@ -233,7 +233,7 @@ public String getRefreshToken() {
 	 * @return The client info.
 	 */
 	public ClientInfo getClientInfo() {
-		return clientInfo;
+		return oAuthRefreshInterceptor.clientInfo;
```

Comment:
the clientInfo is updated for the 2nd request using the access token from authTokenProvider once gotten a 401 return at the first time